indico/modules/events/management/forms.py

```diff
@@ -407,3 +407,33 @@ class ProgramCodesForm(IndicoForm):
     session_block_template = StringField(_('Session blocks'))
     contribution_template = StringField(_('Contributions'))
     subcontribution_template = StringField(_('Subcontributions'))
+
+
+class ImportSourceEventForm(IndicoForm):
+    source_event_url = StringField(_('Event url'))
+
+
+class ImportContentsForm(ImportSourceEventForm):
+    selected_items = IndicoSelectMultipleCheckboxField(_('What to clone'))
+
+    def __init__(self, source_event, target_event, set_defaults=False, **kwargs):
+        cloners = EventCloner.get_cloners(source_event)
+        visible_options = filter(attrgetter('is_visible'), cloners)
+        conflicts = {cloner.name: cloner.has_conflicts(target_event) for cloner in cloners}
+
+        if set_defaults:
+            form_params = {
+                'source_event_url': request.form.get('source_event_url', kwargs.pop('source_event_url', None)),
+                'selected_items': request.form.getlist('selected_items'),
+                'csrf_token': request.form.get('csrf_token')
+            }
+            kwargs['formdata'] = ImmutableMultiDict(form_params)
+
+        super(ImportContentsForm, self).__init__(**kwargs)
+        self.selected_items.choices = [(option.name, option.friendly_name) for option in visible_options]
+        # We disable all cloners that are not available, handle only cloning to new events,
+        # have conflicts with target_event or any of their dependencies has conflicts with target_event
+        self.selected_items.disabled_choices = [option.name for option in visible_options if not option.is_available
+                                                or conflicts[option.name]
+                                                or option.__class__.new_event_only
```

Comment:
```suggestion
                                                or type(option).new_event_only
```

Reply:
or better just use `option.new_event_only`; there's nothing wrong with accessing a class attribute through the instance. and that way a class could in principle override it as an actual `@property`